File: bg/web/src/api/example/manualReward.js
import service from '@/utils/request'

export const grantDesktopReward = (data) => {
  return service({
    url: '/manualReward/grantDesktopReward',
    method: 'post',
    data
  })
}

export const grantReward = (data) => {
  return service({
    url: '/manualReward/grantReward',
    method: 'post',
    data
  })
}

File: bg/server/router/example/manual_reward.go
package example

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ManualRewardRouter struct{}

func (r *ManualRewardRouter) InitManualRewardRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	manualRewardRouter := Router.Group("manualReward").Use(middleware.OperationRecord())
	_ = PublicRouter

	manualRewardRouter.POST("grantDesktopReward", manualRewardApi.GrantDesktopReward)
	manualRewardRouter.POST("grantReward", manualRewardApi.GrantReward)
}
